xayn-ai/src/data/mod.rs

```diff
@@ -87,6 +135,38 @@ impl UserInterests {
             negative: Vec::new(),
         }
     }
+
+    /// Moves all user interests of `other` into `Self`.
+    pub(crate) fn append(&mut self, mut other: Self) {
+        // TODO drop dupes in other
+        append_cois(&mut self.positive, &mut other.positive);
+        append_cois(&mut self.negative, &mut other.negative);
+    }
+
+    /// Re-assigns CoI ids for normalization.
+    pub(crate) fn reassign_ids(&mut self) {
+        reassign_coi_ids(&mut self.positive);
+        reassign_coi_ids(&mut self.negative);
+    }
+}
+
+fn append_cois<C>(locals: &mut Vec<C>, remotes: &mut Vec<C>)
+where
+    C: CoiPoint,
+{
+    // shift remote ids to avoid clashes with local ids
+    let max_local_id = locals.iter().map(|coi| coi.id().0).max().unwrap_or(0);
```

Comment:
when I looked at our current coi implementation, it seemed that coi ids would start from `1`. but yes, better safe than sorry. although in the next PR the handling of coi ids will most probably change anyway so this will hopefully become redundant.

Reply:
i agree, changing the ids will fix this anyways.